cumulusci/core/tests/test_flowrunner.py

```diff
@@ -613,6 +611,30 @@ def change_username(keychain):
 
         save.assert_called_once()
 
+    def test_flowrunner__override_task_with_flow(self):
+        """Test that we return a working FlowCoordinator when a project
+        overrides a flow step that is currently a task with a sub-flow."""
+
+        self.project_config.config["flows"]["test"] = {
+            "description": "Replace a task with a flow",
+            "steps": {1: {"task": "None", "flow": "nested_flow"}},
+        }
+        flow_config = self.project_config.get_flow("test")
+        # No assert, we just don't want an exception thrown
+        FlowCoordinator(self.project_config, flow_config)
+
+    def test_flowrunner__override_flow_with_task(self):
+        """Test that we return a working FlowCoordinator when a project
+        overrides a flow step that is currently a task with a sub-flow."""
+
+        self.project_config.config["flows"]["test"] = {
+            "description": "Replace a task with a flow",
+            "steps": {1: {"task": "pass_name", "flow": "None"}},
+        }
+        flow_config = self.project_config.get_flow("test")
+        # No assert, we just don't want an exception thrown
+        FlowCoordinator(self.project_config, flow_config)
+
```

Comment:
Note that the (already existing) test method `test_init_ambiguous_step()` covers the case where both `task` and `flow` come in with a value of `"None"`.